examples/python/ocr/ocr.py

```diff
@@ -0,0 +1,446 @@
+        layout_type = LayoutType.get_type(type)
+        detections, table, img = [], None, None
+        if layout_type == LayoutType.TABLE:
+            table = self.get_table_markdown(line)
+        elif layout_type == LayoutType.FIGURE:
+            detections = self.get_detections(line)
+            img = line.get("img")  # Currently not in use
+        else:
+            detections = self.get_detections(line)
+        self.add(layout_type, box, detections=detections, table=table, figure=img)
+
+    @staticmethod
+    def get_detections(line: dict[str, Any]) -> list[dict[str, Any]]:
+        detections = []
+        results = line.get("res")
+        if results is not None:
+            for i, result in enumerate(results):
+                text = result.get("text")
+                confidence = result.get("confidence")
+                box = result.get("text_region")
+                x_min, y_min = box[0]
+                x_max, y_max = box[2]
+                new_box = [x_min, y_min, x_max, y_max]
+                detections.append({"id": i, "text": text, "confidence": confidence, "box": new_box})
+        return detections
+
+    # Safely attempt to extract the HTML table from the results
+    @staticmethod
+    def get_table_markdown(line: dict[str, Any]) -> str:
+        try:
+            html_table = line.get("res", {}).get("html")
+            if not html_table:
+                return "No table found."
+
+            dataframes = pd.read_html(html_table)
+            if not dataframes:
+                return "No data extracted from the table."
+
+            markdown_table = dataframes[0].to_markdown()
+            return markdown_table  # type: ignore[no-any-return]
+
+        except Exception as e:
+            return f"Error processing the table: {str(e)}"
+
+
+def process_layout_records(
+    layout: Layout,
+) -> tuple[list[str], list[str], list[rrb.Spatial2DView], list[rrb.Spatial2DView], list[rrb.Spatial2DView]]:
+    paths, detections_paths = [], []
+    zoom_paths: list[rrb.Spatial2DView] = []
+    zoom_paths_figures: list[rrb.Spatial2DView] = []
+    zoom_paths_tables: list[rrb.Spatial2DView] = []
+    zoom_paths_texts: list[rrb.Spatial2DView] = []
+
+    for layout_type in LayoutType:
+        for record in layout.records[layout_type]:
+            record_name = record["name"].title()
+            base_path = f"Image/{layout_type.type.title()}/{record_name}"
+            paths.append(f"-{base_path}/**")
+            detections_paths.append(f"-{base_path}/Detections/**")
+
+            # Log bounding box
+            rr.log(
+                base_path,
+                rr.Boxes2D(
+                    array=record["bounding_box"],
+                    array_format=rr.Box2DFormat.XYXY,
+                    labels=[str(layout_type.type)],
+                    class_ids=[str(layout_type.number)],
+                ),
+                rr.AnyValues(name=record_name),
+                timeless=True,
```

Comment:
deprecated, use `static` instead. Also, does it actually have to be static? Might as well just keep time here, so the log_time/log_tick will tell us when what came in